src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx

```diff
@@ -0,0 +1,403 @@
+import { LatestOnerepScanData } from "../../../../../../../../../../db/tables/onerep_scans";
+import { ExposureCardDataClassLayout } from "../../../../../../../../../components/client/exposure_card/ExposureCardDataClass";
+import confetti from "canvas-confetti";
+import { OnerepScanResultRow } from "knex/types/tables";
+import { toast } from "react-toastify";
+
+export type Props = {
+  data: LatestOnerepScanData;
+  stepDeterminationData: StepDeterminationData;
+  subscriberEmails: string[];
+};
+
+export const RemovalUnderMaintenanceView = (props: Props) => {
+  const l10n = useL10n();
+  const [detailedRemovalGuide, setDetailedRemovalGuide] = useState(false);
+  const [firstScanResultNotResolved, setFirstScanResultNotResolved] = useState(
+    props.data.results[0],
+  );
+  const [isLoadingNextDataBroker, setIsLoadingNextDataBroker] = useState(false);
+  const [fadeState, setFadeState] = useState<"fadeIn" | "fadeOut" | "">("");
+
+  const nextGuidedStep = getNextGuidedStep(
+    props.stepDeterminationData,
+    "DataBrokerManualRemoval",
+  );
+
+  const [nextScanResultNotResolved, setNextScanResultNotResolved] = useState<
+    OnerepScanResultRow | undefined
+  >();
+
+  const updateDataBrokerWithTimeout = () => {
+    setIsLoadingNextDataBroker(true);
+    setFadeState("fadeOut");
+
+    setTimeout(() => {
+      setIsLoadingNextDataBroker(false);
+      void confetti();
+      setFadeState("fadeIn"); // trigger the fade-in for the next scan result
+    }, 2000);
+  };
+
+  async function handleManualRemovalChange() {
+    try {
+      const response = await fetch(
+        `/api/v1/user/scan-result/${firstScanResultNotResolved.onerep_scan_result_id}/resolution`,
+        {
+          method: "POST",
+          credentials: "same-origin",
+        },
+      );
+
+      if (!response.ok) {
+        toast.error("Could not resolve data broker.");
+        console.error(
+          "Could not update next data broker with removal under maintenance status.",
+        );
+        return;
+      }
+
+      // Mark the current scan result as manually resolved
+      firstScanResultNotResolved.manually_resolved = true;
+
+      const nextUnresolved = props.data.results.find(
+        (scanResult) => !scanResult.manually_resolved,
+      );
+
+      // Redirect if no unresolved scan result remains
+      if (!nextUnresolved) {
+        window.location.href = nextGuidedStep.href;
+      } else {
+        setNextScanResultNotResolved(nextUnresolved);
+      }
+
+      // Trigger the timeout update
+      updateDataBrokerWithTimeout();
+    } catch (error) {
+      console.error("Error occurred in handleManualRemovalChange:", error);
+    }
+  }
+
+  // Effect to update the current unresolved scan result when `nextScanResultNotResolved` changes
+  useEffect(() => {
+    if (nextScanResultNotResolved) {
+      setFirstScanResultNotResolved(nextScanResultNotResolved);
+    }
+  }, [nextScanResultNotResolved]);
```

Comment:
This `useEffect` should not be needed. We can directly set the `firstScanResultNotResolved` from within `handleManualRemovalChange` instead of storing the next scan result intermediately in its own state.